src/config/wayland.md

```diff
@@ -0,0 +1,106 @@
+the default session, while Plasma still has a few issues to iron out, so their
+default remains a Xorg session. While running these desktop environments,
+applications built with Gtk+ will automatically choose the Wayland backend, but
+Qt applications might require the proper environment variable in GNOME. This is
+explained [below](#native-applications).
+
+### Stand-alone compositors
+
+Void Linux currently packages the following Wayland compositors:
+
+- Weston: reference compositor for Wayland
+- Sway: an i3-compatible Wayland compositor
+- Wayfire: 3D Wayland compositor
+- Cage: a Wayland kiosk
+
+### Native applications
+
+Qt5 based applications require the installation of the `qt5-wayland` package and
+setting the environment variable `QT_QPA_PLATFORM=wayland-egl` to enable their
+Wayland backend. Some KDE specific applications might also require the
+installation of the `kwayland` package. Gtk+ applications should do it
+automatically, but setting `GDK_BACKEND=wayland` can force the Wayland backend
+(however, this can break applications such as Chromium).
+
+#### Web browsers
+
+Mozilla Firefox ships with a Wayland backend which is disabled by default.
+Launching Firefox with `MOZ_ENABLE_WAYLAND=1` is required to enable it.
+
+The Midori browser, which has a Gtk+ interface, should also work on Wayland.
+
+Qt5 based browsers, such as qutebrowser, also work on Wayland.
+
+#### Terminal emulators
+
+- Alacritty (it's the standard terminal emulator for Sway)
+- Kitty
+- Sakura
+
+#### Media applications
+
+Both mpv and VLC work on Wayland, with hardware acceleration (if available and
+enabled).
+
+For image viewing, imv works natively on Wayland.
+
+#### General desktop utilities
+
+- For screenshots: `grim` and `slurp`. `slurp` is required to define a geometry.
+- For accessing the clipboard: `wl-clipboard`.
+- For displaying notifications (notification daemon): `mako`.
+- For launching applications:
+   - wofi
+   - bemenu
+   - dmenu-wayland
+
+#### Running X applications inside Wayland
+
+If an application still doesn't support Wayland, it can still be run in a
+Wayland context. XWayland is an X server that bridges this gap for most Wayland
+compositors, and is installed as a dependency for most of them. Its package is
+`xorg-server-xwayland`. For Weston, the correct package is `weston-xwayland`.
+
+## Configuration
+
+The Wayland API uses the `XDG_RUNTIME_DIR` environment variable to determine the
+directory the Wayland socket. If this environment isn't already set in your
+setup, you can add this snippet to your shell profile or to a script that
+starts-up your Wayland compositor of choice. This shouldn't be necessary with
+any of the desktop environments.
+
+```
+if test -z "${XDG_RUNTIME_DIR}"; then
+    export XDG_RUNTIME_DIR=/tmp/${UID}-runtime-dir
+    if ! test -d "${XDG_RUNTIME_DIR}"; then
+        mkdir -p "${XDG_RUNTIME_DIR}"
+        chmod 0700 "${XDG_RUNTIME_DIR}"
+    fi
+fi
+```
```

Comment:
Although it appears [here](https://wayland.freedesktop.org/building.html), @Duncaen pointed out that this snippet has some flaws, but I don't remember the details now.

Reply:
As you mentioned below, elogind also sets it up, so it's definitely a solution. However, I feel like including the script could make life easier for some people (as it did for me, as a matter of fact). If it has issues, though, it's probably a good idea to remove it, seeing as we want the Void documentation to be properly curated.

Reply:
I'm also thinking of including the `XDG_SESSION_TYPE` variable, and the fact that you're supposed to set it to `wayland`. Do you know any applications that use its value? My own scripts do, and I feel like it's the "correct" thing to do, but I have no idea if it's necessary.

Reply:
`$XDG_RUNTIME_DIR` is usually on `/run` because only privileged users can create directories there opposed to `/tmp` where this could haven been maliciously created before your shell script runs on login.

Reply:
Considering this is a possible security hazard, do you think it should be documented as such, or fully removed?

Reply:
i would prefer the documentation include a (more) secure way of setting up `XDG_RUNTIME_DIR`, since (a) it's necessary for Wayland, and (b) if people aren't provided with a more secure way, they'll instead use whatever 'solution' seems to 'work' - whether that's something they've put together themselves, or something they've just copied and pasted from somewhere else.

Reply:
The right solution is to use elogind.

Reply:
All right, I will leave elogind as the only option. Do you think it's best to explain why elogind is necessary (it has to create the directory and set `XDG_RUNTIME_DIR`), or should I just mention that elogind is necessary for wayland and leave it at that?

Reply:
Fixed.